src/core/lib/surface/validate_metadata.cc

```diff
@@ -67,6 +67,10 @@ grpc_error* grpc_validate_header_key_is_legal(const grpc_slice& slice) {
     return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
         "Metadata keys cannot be zero length");
   }
+  if (GRPC_SLICE_LENGTH(slice) > UINT32_MAX) {
```

Comment:
can you explain why this branch is added?

Reply:
Yes - consider the following (now removed) code in hpack_encoder.cc:

"
size_t len_val = wire_value_length(value);
GPR_ASSERT(len_val <= UINT32_MAX);
"

This is executed by methods like: emit_lithdr_(inc/no)idx[_v] for any header that ended up taking that path.

However:
1) If a metadata key/value pair is not provided by the user, we should do a debug assert and not a regular assert, since we control what metadata core and our filters use.
2) If a metadata key/value is provided by the user, we should in that case do the length check as a regular assert - but it only needs to be done for user provided metadata.

Hence, this check.

Reply:
Note that validate_metadata is called in prepare_application_metadata, which handles user provided metadata.

Reply:
Ah I see. Makes sense.